server/ai_mediaserver.go

```diff
@@ -837,7 +837,10 @@ func (ls *LivepeerServer) CreateWhip(server *media.WHIPServer) http.Handler {
 						clog.Errorf(ctx, "Panic in stream close event routine: %s", r)
 					}
 				}()
-				whipConn.AwaitClose()
+				err := whipConn.AwaitClose()
```

Comment:
@j0sh We also have an `AwaitClose` call [here](https://github.com/livepeer/go-livepeer/blob/e71c451c079cd282f46844d91ff89fd0272ce935/server/ai_mediaserver.go#L888). Should we tidy that up to only call it once? It's a bit confusing right now having shutdown type logic in two places and we only want to log these errors once. I'll still merge this since I think it'll be useful to have in the meantime while you're on leave.

Reply:
Perhaps, but this allows us to have more accurate timestamps for event and error reporting. For example, if the WHIP connection closes before orch selection completes then we can know that.

Otherwise the sequencing of events might get a little confusing when we are dealing with tight timings (which is usually the case with startup errors, for example)

Also, having an AwaitClose after orch selection is also why WHIP was not affected by the race condition from https://github.com/livepeer/go-livepeer/pull/3483

So we kind of need the await in both places.